liblwgeom/lwgeom.c

```diff
@@ -2395,3 +2396,59 @@ lwgeom_is_trajectory(const LWGEOM *geom)
 	return lwline_is_trajectory((LWLINE*)geom);
 }
 
+static uint32_t
+bits_for_precision(int32_t significant_digits)
+{
+	return (uint32_t) ceil(significant_digits / log10(2));
+}
+
+static inline
+double mask_double(double d, uint64_t mask)
+{
+	uint64_t* double_bits = (uint64_t*) (&d);
+
+	(*double_bits) &= mask;
+
+	return *((double*) double_bits);
+}
+
+static double trim_preserve_decimal_digits(double d, int32_t decimal_digits)
+{
+	if (d==0)
+		return 0;
+
+	int digits_left_of_decimal = (int) (1 + log10(fabs(d)));
+	uint32_t bits_needed = bits_for_precision(abs(decimal_digits) + digits_left_of_decimal);
```

Comment:
http://www.cplusplus.com/reference/clibrary/cmath/frexp/

Reply:
@mguzelevich Could you elaborate on how to use that here?